File: test_minimal.py
#!/usr/bin/env python
"""Minimal test matching OCaml version."""

from test_deterministic import load_random_tape, add_tile_deterministic
from game import Game2048

# Initialize
Game2048._init_tables()

# Load tape
events, _ = load_random_tape("test_controlled.txt")
tape_data = (events, 0)

# Initial empty board
board = 0

# Add two initial tiles
board, tape_data = add_tile_deterministic(board, tape_data)
print(f"After tile 1: {board:016x}")

board, tape_data = add_tile_deterministic(board, tape_data)
print(f"After tile 2: {board:016x}")

# Make a move
board, score, _ = Game2048.move_left(board)
print(f"After left move: {board:016x}")

# Add another tile
board, tape_data = add_tile_deterministic(board, tape_data)
print(f"After tile 3: {board:016x}")

print(f"Tape position used: {tape_data[1]}")

File: test_minimal.ml
open Gp_2048_lib.Game
open Gp_2048_lib.Random_tape

let () =
  (* Simple test with controlled random tape *)
  let tape = load_random_tape "test_controlled.txt" in
  
  (* Initial empty board *)
  let board = ref empty_board in
  
  (* Add two initial tiles *)
  board := add_random_tile_from_tape !board tape;
  Printf.printf "After tile 1: %016Lx\n" !board;
  
  board := add_random_tile_from_tape !board tape;
  Printf.printf "After tile 2: %016Lx\n" !board;
  
  (* Make a move *)
  board := move_left !board;
  Printf.printf "After left move: %016Lx\n" !board;
  
  (* Add another tile *)
  board := add_random_tile_from_tape !board tape;
  Printf.printf "After tile 3: %016Lx\n" !board;
  
  Printf.printf "Tape position used: %d\n" tape.index